video/external/src/backend.rs

```diff
@@ -25,154 +24,52 @@ enum ProxyOrStream {
-            ),
-            (os, arch) => return Err(format!("Unsupported OS/arch: {}/{}", os, arch).into()),
-        };
-
-        Ok(OpenH264Data {
-            local_filenames,
-            download_filename,
-            download_sha256,
-        })
+    #[cfg(feature = "openh264")]
+    pub fn load_openh264(directory: &Path) -> Result<OpenH264Codec, Box<dyn std::error::Error>> {
+        crate::decoder::openh264::H264Decoder::load_openh264(directory)
     }
 
-    fn download_openh264(
-        openh264_data: &OpenH264Data,
-        directory: &Path,
-    ) -> Result<PathBuf, Box<dyn std::error::Error>> {
-        // See the license at: https://www.openh264.org/BINARY_LICENSE.txt
-        const URL_BASE: &str = "http://ciscobinary.openh264.org/";
-        const URL_SUFFIX: &str = ".bz2";
-
-        let (filename, sha256sum) = (
-            openh264_data.download_filename,
-            openh264_data.download_sha256,
-        );
-
-        std::fs::create_dir_all(directory)?;
-        let filepath = directory.join(filename);
-
-        // If the binary doesn't exist in the expected location, download it.
-        if !filepath.is_file() {
-            let url = format!("{}{}{}", URL_BASE, filename, URL_SUFFIX);
-            let response = reqwest::blocking::get(url)?;
-            let mut bzip2_reader = BzDecoder::new(response);
-
-            let mut tempfile = tempfile::NamedTempFile::with_prefix_in(filename, directory)?;
-            copy(&mut bzip2_reader, &mut tempfile)?;
-            // Let's assume that if this fails, it's because another process has already put it there
-            // and loaded it, therefore it can't be overwritten (on Windows at least), but in the end,
-            // all's fine - the hash will still be checked before attempting to load the library.
-            let _ = tempfile.persist(&filepath);
+    fn make_decoder(&mut self) -> Result<Box<dyn VideoDecoder>, Error> {
+        #[cfg(feature = "openh264")]
+        if let Some(h264_codec) = self.openh264_codec.as_ref() {
+            let decoder = Box::new(crate::decoder::openh264::H264Decoder::new(h264_codec));
+            return Ok(decoder);
         }
 
-        // Regardless of whether the library was already there, or we just downloaded it, let's check the MD5 hash.
-        let mut sha256 = Sha256::new();
-        copy(&mut File::open(filepath.clone())?, &mut sha256)?;
-        let sha256digest = sha256.finalize();
-        let result: [u8; 32] = sha256digest.into();
-
-        if result[..] != hex::decode(sha256sum)?[..] {
-            let size = filepath.metadata().map(|f| f.len()).unwrap_or_default();
-            return Err(format!(
-                "SHA256 checksum mismatch for {filename}; expected {sha256sum}, found {sha256digest:x} (with a size of {size} bytes)",
-            )
-            .into());
-        }
-
-        Ok(filepath)
+        #[allow(unreachable_code)]
+        return Err(Error::DecoderError("No OpenH264".into()));
```

Comment:
```suggestion
        Err(Error::DecoderError("No OpenH264".into()))
```

Reply:
Right, nice, thanks! In some earlier incarnation of this code, this wasn't possible without clippy freaking out.